app/src/main/java/com/framgia/moviedbtraining/movieDetails/MovieDetailsPresenter.kt

```diff
@@ -7,6 +7,14 @@ import com.framgia.moviedbtraining.utils.GeneralUtil
 
 class MovieDetailsPresenter(
     private val mViewModel: MovieDetailsContract.ViewModel, movieId: Int) : MovieDetailsContract.Presenter {
+  override fun rateMovie(mId: Int, toFloat: Float) {
+    if ((!GeneralUtil.isNetworkAvailable(App.self()))) {
```

Comment:
should change GeneralUtil.isNetworkAvailable with App.self() in self method, no need to pass here